x-pack/plugins/apm/public/components/shared/TransactionActionMenu/TransactionActionMenu.tsx

```diff
@@ -65,9 +68,12 @@ export class TransactionActionMenu extends React.Component<Props, State> {
   };
 
   public getInfraActions(transaction: Transaction) {
-    const hostName = get(transaction, 'context.system.hostname');
-    const podId = get(transaction, 'kubernetes.pod.uid');
-    const containerId = get(transaction, 'container.id');
+    const hostName = idx(transaction, _ => _.context.system.hostname);
+    const podId = idx(transaction as TransactionV2, _ => _.kubernetes.pod.uid);
+    const containerId = idx(
+      transaction as TransactionV2,
+      _ => _.docker.container.id
+    );
```

Comment:
Rather than using the `as` syntax in TS, could we check the `version` key so we get the discriminated union stuff?

e.g.
```js
const podId = transaction.version === 'v2' ? idx(transaction, _ => _.kubernetes.pod.uid) : undefined;
```

Reply:
The `as` syntax also felt a bit fishy to me, but I think it's better than the alternatives:
 - we don't really care whether it's v1 or v2, so okay to cast it to v2, and then it'll just resolve to `null` in runtime if it doesn't exist
 - we are getting rid of v1 and v2 anyway very soon (7.0), so the checks we are adding now will be removed in a couple of weeks anyway